sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobs.cs

```diff
@@ -0,0 +1,288 @@
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() =>
+            ((this.IncludeCopyMetadata     ? 0b00001 : 0) +
+             (this.IncludeDeletedBlobs     ? 0b00010 : 0) +
+             (this.IncludeMetadata         ? 0b00100 : 0) +
+             (this.IncludeSnapshots        ? 0b01000 : 0) +
+             (this.IncludeUncommittedBlobs ? 0b10000 : 0)) ^
+            (this.Prefix?.GetHashCode() ?? 0) ^
+            this.PageSizeHint.GetHashCode();
+
+        /// <summary>
+        /// Check if two GetBlobsOptions instances are equal.
+        /// </summary>
+        /// <param name="left">The first instance to compare.</param>
+        /// <param name="right">The second instance to compare.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public static bool operator ==(GetBlobsOptions left, GetBlobsOptions right) =>
+            left.Equals(right);
+
+        /// <summary>
+        /// Check if two GetBlobsOptions instances are not equal.
+        /// </summary>
+        /// <param name="left">The first instance to compare.</param>
+        /// <param name="right">The second instance to compare.</param>
+        /// <returns>True if they're not equal, false otherwise.</returns>
+        public static bool operator !=(GetBlobsOptions left, GetBlobsOptions right) =>
+            !(left == right);
+
+        /// <summary>
+        /// Check if two GetBlobsOptions instances are equal.
+        /// </summary>
+        /// <param name="obj">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public bool Equals(GetBlobsOptions other) =>
+            this.IncludeCopyMetadata == other.IncludeCopyMetadata &&
+            this.IncludeDeletedBlobs == other.IncludeDeletedBlobs &&
+            this.IncludeMetadata == other.IncludeMetadata &&
+            this.IncludeSnapshots == other.IncludeSnapshots &&
+            this.IncludeUncommittedBlobs == other.IncludeUncommittedBlobs &&
+            this.Prefix == other.Prefix &&
+            this.PageSizeHint == other.PageSizeHint;
+    }
+
+    internal class GetBlobsAsyncCollection : StorageAsyncCollection<BlobItem>
+    {
+        private readonly BlobContainerClient _client;
+        private readonly GetBlobsOptions? _options;
+
+        public GetBlobsAsyncCollection(
+            BlobContainerClient client,
+            GetBlobsOptions? options,
+            CancellationToken cancellationToken)
+            : base(cancellationToken)
+        {
+            this._client = client;
+            this._options = options;
+            this.PageSizeHint = options?.PageSizeHint;
+        }
+
+        protected override async Task<Page<BlobItem>> GetNextPageAsync(
+            string continuationToken,
+            bool isAsync,
+            CancellationToken cancellationToken = default)
+        {
+            var task = this._client.GetBlobsAsync(
```

Comment:
I really think we should stop using var. It's impossible to review code without knowing return types of some of these calls.